--- random-trait-0.1.1/Cargo.toml	1970-01-01T00:00:00+00:00
+++ random-trait-0.1.1/Cargo.toml	2024-09-14T12:56:06.285124+00:00
@@ -14,9 +14,8 @@
 name = "random-trait"
 version = "0.1.1"
 authors = ["Elichai Turkel <elichai.turkel@gmail.com>"]
-include = ["**/*.rs", "Cargo.toml"]
+include = ["**/*.rs", "Cargo.toml", "LICENSE-*"]
 description = "Rust library for a random trait meant to produce random generic types"
-readme = "README.md"
 keywords = ["rand", "random", "random-rs", "rng"]
 categories = ["algorithms", "no-std", "development-tools::testing", "cryptography"]
 license = "MIT/Apache-2.0"
