diff --git a/pyproject.toml b/pyproject.toml
index 08191a1..7027ef6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,7 +11,7 @@ shazamctl = "shazamctl.cli:app"
 
 [tool.poetry.dependencies]
 python = "^3.10"
-typer = "^0.4.1"
+typer = "^0.13"
 shazamapi = {url = "https://github.com/notpushkin/ShazamAPI/archive/refs/heads/refactor.zip"}
 PyAudio = { version = "^0.2.11", optional = true }
 

