diff --git a/yazi-core/Cargo.toml b/yazi-core/Cargo.toml
index 5acc011..15d131f 100644
--- a/yazi-core/Cargo.toml
+++ b/yazi-core/Cargo.toml
@@ -25,7 +25,7 @@ bitflags      = "2.6.0"
 crossterm     = "0.27.0"
 dirs          = "5.0.1"
 futures       = "0.3.30"
-notify        = { git = "https://github.com/notify-rs/notify.git", rev = "96dec74316a93bed6eec9db177b233e6e017275e", default-features = false, features = [ "macos_fsevent" ] }
+notify        = { path = "../vendor/notify" }
 parking_lot   = "0.12.3"
 ratatui       = "0.27.0"
 scopeguard    = "1.2.0"

