diff --git a/generate-symbols.py b/generate-symbols.py
index 5b0ee95..a32136f 100755
--- a/generate-symbols.py
+++ b/generate-symbols.py
@@ -17,9 +17,10 @@ for rng in ranges:
             character = chr(i)
             name = unicodedata.name(character)
             print(
-                (character + " " + name.lower()).ljust(60, " "),
-                hex(i)[2:].upper().rjust(6, " "),
-                rng[2],
+                hex(i)[2:].upper(),
+                name.lower(),
+                "(" + rng[2] + ")",
+                f"\x00icon\x1f<span color=\"white\">{character}</span>",
             )
         except:
             continue

