diff --git a/man/hwatch.1 b/man/hwatch.1
index fffb607..24814f9 100644
--- a/man/hwatch.1
+++ b/man/hwatch.1
@@ -211,10 +211,10 @@ F3
 Display \f[I]Stdout\f[R] and \f[I]Stderr\f[R].
 .TP
 +
-Increase interval by 0.5 seconds.
+Increase interval by 0.1 seconds.
 .TP
 \-
-Decrease interval by 0.5 seconds (As long as it\[cq]s positive).
+Decrease interval by 0.1 seconds (As long as it\[cq]s positive).
 .TP
 Tab
 Switch the target(\f[I]history\f[R] or \f[I]watch\f[R] pad).
diff --git a/man/man.md b/man/man.md
index 45b8557..94bf8b9 100644
--- a/man/man.md
+++ b/man/man.md
@@ -247,11 +247,11 @@ F3
 
 \+
 
-:   Increase interval by 0.5 seconds.
+:   Increase interval by 0.1 seconds.
 
 \-
 
-:   Decrease interval by 0.5 seconds (As long as it's positive).
+:   Decrease interval by 0.1 seconds (As long as it's positive).
 
 Tab
 
diff --git a/src/app.rs b/src/app.rs
index 1c1b8d1..1261e7c 100644
--- a/src/app.rs
+++ b/src/app.rs
@@ -606,14 +606,14 @@ impl<'a> App<'a> {
     ///
     fn increase_interval(&mut self) {
         let cur_interval = *self.interval.read().unwrap();
-        self.set_interval(cur_interval + 0.5);
+        self.set_interval(cur_interval + 0.1);
     }
 
     ///
     fn decrease_interval(&mut self) {
         let cur_interval = *self.interval.read().unwrap();
-        if cur_interval > 0.5 {
-            self.set_interval(cur_interval - 0.5);
+        if cur_interval >= 0.1 {
+            self.set_interval(cur_interval - 0.1);
         }
     }
 
diff --git a/src/keymap.rs b/src/keymap.rs
index 169dc91..42c8803 100644
--- a/src/keymap.rs
+++ b/src/keymap.rs
@@ -540,8 +540,8 @@ pub fn get_input_action_description(input_action: InputAction) -> String {
         InputAction::ToggleHistorySummary => "Toggle history summary".to_string(),
 
         // Interval
-        InputAction::IntervalPlus => "Interval +0.5sec".to_string(),
-        InputAction::IntervalMinus => "Interval -0.5sec".to_string(),
+        InputAction::IntervalPlus => "Interval +0.1sec".to_string(),
+        InputAction::IntervalMinus => "Interval -0.1sec".to_string(),
 
         // Command
         InputAction::ChangeFilterMode => "Change filter mode".to_string(),

