diff --git a/src/app.rs b/src/app.rs
index 4512092..41601ea 100644
--- a/src/app.rs
+++ b/src/app.rs
@@ -1308,19 +1308,6 @@ impl<'a> App<'a> {
         let _ = self.tx.send(AppEvent::Redraw);
     }
 
-    ///
-    pub fn show_help_banner(&mut self, visible: bool) {
-        self.header_area.set_banner(
-            if visible {
-                "Display help with h key!"
-            } else {
-                ""
-            }
-            .to_string(),
-        );
-        let _ = self.tx.send(AppEvent::Redraw);
-    }
-
     ///
     fn input_key_up(&mut self) {
         match self.window {
diff --git a/src/header.rs b/src/header.rs
index 3aea75f..c14adfc 100644
--- a/src/header.rs
+++ b/src/header.rs
@@ -121,10 +121,6 @@ impl<'a> HeaderArea<'a> {
         self.line_number = line_number;
     }
 
-    pub fn set_banner(&mut self, banner: String) {
-        self.banner = banner;
-    }
-
     pub fn set_ansi_color(&mut self, ansi_color: bool) {
         self.ansi_color = ansi_color;
     }
diff --git a/src/main.rs b/src/main.rs
index 14b9de0..39e2cb1 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -182,14 +182,6 @@ fn build_app() -> clap::Command {
                 .long("line-number"),
         )
         // exec flag.
-        //     [--no-help-banner]
-        .arg(
-            Arg::new("no_help_banner")
-                .help("hide the \"Display help with h key\" message")
-                .long("no-help-banner")
-                .action(ArgAction::SetTrue),
-        )
-        // exec flag.
         //     [-x,--exec]
         .arg(
             Arg::new("exec")
@@ -434,8 +426,7 @@ fn main() {
             .set_diff_mode(diff_mode)
             .set_only_diffline(matcher.get_flag("diff_output_only"))
 
-            .set_show_ui(!matcher.get_flag("no_title"))
-            .set_show_help_banner(!matcher.get_flag("no_help_banner"));
+            .set_show_ui(!matcher.get_flag("no_title"));
 
         // Set logfile
         if let Some(logfile) = logfile {
diff --git a/src/view.rs b/src/view.rs
index 20c9b86..6eb7758 100644
--- a/src/view.rs
+++ b/src/view.rs
@@ -35,7 +35,6 @@ pub struct View {
     mouse_events: bool,
     color: bool,
     show_ui: bool,
-    show_help_banner: bool,
     line_number: bool,
     output_mode: OutputMode,
     diff_mode: DiffMode,
@@ -54,7 +53,6 @@ impl View {
             mouse_events: false,
             color: false,
             show_ui: true,
-            show_help_banner: true,
             line_number: false,
             output_mode: OutputMode::Output,
             diff_mode: DiffMode::Disable,
@@ -98,11 +96,6 @@ impl View {
         self
     }
 
-    pub fn set_show_help_banner(mut self, show_help_banner: bool) -> Self {
-        self.show_help_banner = show_help_banner;
-        self
-    }
-
     pub fn set_line_number(mut self, line_number: bool) -> Self {
         self.line_number = line_number;
         self
@@ -175,7 +168,6 @@ impl View {
 
         app.show_history(self.show_ui);
         app.show_ui(self.show_ui);
-        app.show_help_banner(self.show_help_banner);
 
         app.set_tab_size(self.tab_size);
 
