diff --git a/hyprland.spec b/hyprland.spec
index 881bf28..914ee97 100644
--- a/hyprland.spec
+++ b/hyprland.spec
@@ -1,20 +1,23 @@
 Name:           hyprland
-Version:        0.39.1
+Version:        0.43.0
 Release:        %autorelease
 Summary:        Dynamic tiling Wayland compositor that doesn't sacrifice on its looks
 
 # hyprland: BSD-3-Clause
-# subprojects/hyprland-protocols: BSD-3-Clause
-# subprojects/wlroots-hyprland: MIT
-# subproject/udis86: BSD-2-Clause
-# protocols/ext-workspace-unstable-v1.xml: HPND-sell-variant
-# protocols/wlr-foreign-toplevel-management-unstable-v1.xml: HPND-sell-variant
-# protocols/wlr-layer-shell-unstable-v1.xml: HPND-sell-variant
-# protocols/idle.xml: LGPL-2.1-or-later
-License:        BSD-3-Clause AND MIT AND BSD-2-Clause AND HPND-sell-variant AND LGPL-2.1-or-later
+# ./subprojects/udis86: BSD-2-Clause
+# ./protocols/kde-server-decoration.xml: LGPL-2.1-or-later
+# ./protocols/wayland-drm.xml: HPND-sell-variant and/or ntp_disclaimer
+# ./protocols/wlr-data-control-unstable-v1.xml: HPND-sell-variant and/or ntp_disclaimer
+# ./protocols/wlr-foreign-toplevel-management-unstable-v1.xml: HPND-sell-variant and/or ntp_disclaimer
+# ./protocols/wlr-gamma-control-unstable-v1.xml: HPND-sell-variant and/or ntp_disclaimer
+# ./protocols/wlr-layer-shell-unstable-v1.xml: HPND-sell-variant and/or ntp_disclaimer
+# ./protocols/wlr-output-management-unstable-v1.xml: HPND-sell-variant and/or ntp_disclaimer
+License:        BSD-3-Clause AND BSD-2-Clause AND LGPL-2.1-or-later AND HPND-sell-variant
 URL:            https://github.com/hyprwm/Hyprland
 Source:         %{url}/releases/download/v%{version}/source-v%{version}.tar.gz
 
+Patch:          ime-decorations.diff
+
 # https://fedoraproject.org/wiki/Changes/EncourageI686LeafRemoval
 ExcludeArch:    %{ix86}
 
@@ -23,6 +26,7 @@ BuildRequires:  gcc-c++
 BuildRequires:  jq
 BuildRequires:  meson
 
+BuildRequires:  pkgconfig(aquamarine)
 BuildRequires:  pkgconfig(cairo)
 BuildRequires:  pkgconfig(egl)
 BuildRequires:  pkgconfig(gbm)
@@ -31,22 +35,25 @@ BuildRequires:  pkgconfig(hwdata)
 BuildRequires:  pkgconfig(hyprcursor)
 BuildRequires:  pkgconfig(hyprland-protocols)
 BuildRequires:  pkgconfig(hyprlang)
+BuildRequires:  pkgconfig(hyprutils)
+BuildRequires:  pkgconfig(hyprwayland-scanner)
 BuildRequires:  pkgconfig(libdisplay-info)
 BuildRequires:  pkgconfig(libdrm)
-BuildRequires:  pkgconfig(libinput) >= 1.23.0
-BuildRequires:  pkgconfig(libliftoff) >= 0.4.1
+BuildRequires:  pkgconfig(libinput)
+BuildRequires:  pkgconfig(libliftoff)
 BuildRequires:  pkgconfig(libseat)
 BuildRequires:  pkgconfig(libudev)
 BuildRequires:  pkgconfig(pango)
 BuildRequires:  pkgconfig(pangocairo)
-BuildRequires:  pkgconfig(pixman-1) >= 0.42.0
+BuildRequires:  pkgconfig(pixman-1)
 BuildRequires:  pkgconfig(tomlplusplus)
+BuildRequires:  pkgconfig(uuid)
 BuildRequires:  pkgconfig(wayland-client)
 BuildRequires:  pkgconfig(wayland-protocols)
-BuildRequires:  pkgconfig(wayland-scanner)
-BuildRequires:  pkgconfig(wayland-server) >= 1.22.0
+BuildRequires:  pkgconfig(wayland-server)
 BuildRequires:  pkgconfig(xcb-composite)
 BuildRequires:  pkgconfig(xcb-dri3)
+BuildRequires:  pkgconfig(xcb-errors)
 BuildRequires:  pkgconfig(xcb-ewmh)
 BuildRequires:  pkgconfig(xcb-icccm)
 BuildRequires:  pkgconfig(xcb-present)
@@ -58,22 +65,18 @@ BuildRequires:  pkgconfig(xcb-util)
 BuildRequires:  pkgconfig(xcb-xfixes)
 BuildRequires:  pkgconfig(xcb-xinput)
 BuildRequires:  pkgconfig(xcb)
+BuildRequires:  pkgconfig(xcursor)
 BuildRequires:  pkgconfig(xkbcommon)
 BuildRequires:  pkgconfig(xwayland)
 
-# Upstream insists on always building against very current snapshots of
-# wlroots, and doesn't provide a method for building against a system copy.
-# https://github.com/hyprwm/Hyprland/issues/302
-Provides:       bundled(wlroots-hyprland) = 0.18.0~1.git611a4f2
-
 # udis86 is packaged in Fedora, but the copy bundled here is actually a
 # modified fork.
 Provides:       bundled(udis86) = 1.7.2^1.git5336633
 
 Requires:       xorg-x11-server-Xwayland%{?_isa}
 Requires:       xdg-desktop-portal%{?_isa}
-Requires:       libdrm%{?_isa} >= 2.4.120
-Requires:       hyprcursor%{?_isa} >= 0.1.7
+Requires:       hyprcursor%{?_isa} >= 0.1.9
+Requires:       hyprutils%{?_isa} >= 0.2.1
 
 # Both are used in the default configuration
 Recommends:     kitty
@@ -87,8 +90,8 @@ Recommends:     (qt5-qtwayland if qt5-qtbase-gui)
 Recommends:     (qt6-qtwayland if qt6-qtbase-gui)
 
 %description
-Hyprland is a dynamic tiling Wayland compositor based on wlroots that doesn't
-sacrifice on its looks. It supports multiple layouts, fancy effects, has a
+Hyprland is a dynamic tiling Wayland compositor that doesn't sacrifice
+on its looks. It supports multiple layouts, fancy effects, has a
 very flexible IPC model allowing for a lot of customization, a powerful
 plugin system and more.
 
@@ -98,32 +101,34 @@ Requires:       %{name}%{?_isa} = %{version}-%{release}
 Requires:       cmake
 Requires:       cpio
 Requires:       gcc-c++
-Requires:       meson
 Requires:       ninja-build
+Requires:       pkgconfig(aquamarine)
 Requires:       pkgconfig(cairo)
 Requires:       pkgconfig(egl)
 Requires:       pkgconfig(gbm)
 Requires:       pkgconfig(glesv2)
 Requires:       pkgconfig(hwdata)
 Requires:       pkgconfig(hyprcursor)
-Requires:       pkgconfig(hyprland-protocols)
 Requires:       pkgconfig(hyprlang)
+Requires:       pkgconfig(hyprutils)
+Requires:       pkgconfig(hyprwayland-scanner)
 Requires:       pkgconfig(libdisplay-info)
 Requires:       pkgconfig(libdrm)
-Requires:       pkgconfig(libinput) >= 1.23.0
-Requires:       pkgconfig(libliftoff) >= 0.4.1
+Requires:       pkgconfig(libinput)
+Requires:       pkgconfig(libliftoff)
 Requires:       pkgconfig(libseat)
 Requires:       pkgconfig(libudev)
 Requires:       pkgconfig(pango)
 Requires:       pkgconfig(pangocairo)
-Requires:       pkgconfig(pixman-1) >= 0.42.0
+Requires:       pkgconfig(pixman-1)
 Requires:       pkgconfig(tomlplusplus)
+Requires:       pkgconfig(uuid)
 Requires:       pkgconfig(wayland-client)
 Requires:       pkgconfig(wayland-protocols)
-Requires:       pkgconfig(wayland-scanner)
-Requires:       pkgconfig(wayland-server) >= 1.22.0
+Requires:       pkgconfig(wayland-server)
 Requires:       pkgconfig(xcb-composite)
 Requires:       pkgconfig(xcb-dri3)
+Requires:       pkgconfig(xcb-errors)
 Requires:       pkgconfig(xcb-ewmh)
 Requires:       pkgconfig(xcb-icccm)
 Requires:       pkgconfig(xcb-present)
@@ -135,6 +140,7 @@ Requires:       pkgconfig(xcb-util)
 Requires:       pkgconfig(xcb-xfixes)
 Requires:       pkgconfig(xcb-xinput)
 Requires:       pkgconfig(xcb)
+Requires:       pkgconfig(xcursor)
 Requires:       pkgconfig(xkbcommon)
 Requires:       pkgconfig(xwayland)
 Recommends:     git-core
@@ -146,34 +152,29 @@ Recommends:     git-core
 %prep
 %autosetup -n %{name}-source -p1
 rm -rf subprojects/{tracy,hyprland-protocols}
+# don't run generateVersion.sh, release tarballs have pregenerated version.h
+sed -i '/version_h/d' meson.build
 
 cp -p subprojects/udis86/LICENSE LICENSE-udis86
-cp -p subprojects/wlroots-hyprland/LICENSE LICENSE-wlroots
 
 
 %build
-%meson \
-       -Dwlroots-hyprland:examples=false \
-       -Dwlroots-hyprland:xcb-errors=disabled \
-       -Dwlroots-hyprland:werror=false
+%meson
 %meson_build
 
 
 %install
-%meson_install --skip-subprojects wlroots-hyprland
-mkdir -p %{buildroot}%{bash_completions_dir}
-mv %{buildroot}%{_datadir}/bash-completion/hyprctl %{buildroot}%{bash_completions_dir}/hyprctl
-mv %{buildroot}%{_datadir}/bash-completion/hyprpm %{buildroot}%{bash_completions_dir}/hyprpm
+%meson_install
 rm -rf %{buildroot}%{_includedir}/%{name}
 rm -rf %{buildroot}%{_datadir}/pkgconfig/%{name}.pc
 
 
 %files
-%license LICENSE LICENSE-udis86 LICENSE-wlroots
+%license LICENSE LICENSE-udis86
 %{_bindir}/hyprctl
 %{_bindir}/Hyprland
 %{_bindir}/hyprpm
-%{_datadir}/%{name}/
+%{_datadir}/hypr/
 %{_datadir}/wayland-sessions/%{name}.desktop
 %{_datadir}/xdg-desktop-portal/%{name}-portals.conf
 %{_mandir}/man1/hyprctl.1*
diff --git a/ime-decorations.diff b/ime-decorations.diff
new file mode 100644
index 0000000..baf777b
--- /dev/null
+++ b/ime-decorations.diff
@@ -0,0 +1,25 @@
+diff --git a/src/render/Renderer.cpp b/src/render/Renderer.cpp
+index 7b29eb77..9f6c6748 100644
+--- a/src/render/Renderer.cpp
++++ b/src/render/Renderer.cpp
+@@ -791,11 +791,14 @@ void CHyprRenderer::renderIMEPopup(CInputPopup* pPopup, CMonitor* pMonitor, time
+ 
+     const auto  SURF = pPopup->getSurface();
+ 
+-    renderdata.blur     = false;
+-    renderdata.surface  = SURF;
+-    renderdata.decorate = false;
+-    renderdata.w        = SURF->current.size.x;
+-    renderdata.h        = SURF->current.size.y;
++    renderdata.surface   = SURF;
++    renderdata.decorate  = false;
++    renderdata.w         = SURF->current.size.x;
++    renderdata.h         = SURF->current.size.y;
++
++    renderdata.blur      = true;
++    g_pHyprOpenGL->m_RenderData.discardMode |= DISCARD_ALPHA;
++    g_pHyprOpenGL->m_RenderData.discardOpacity = 0.4;
+ 
+     SURF->breadthfirst([](SP<CWLSurfaceResource> s, const Vector2D& offset, void* data) { renderSurface(s, offset.x, offset.y, data); }, &renderdata);
+ }
+
diff --git a/sources b/sources
index cfad131..3624e45 100644
--- a/sources
+++ b/sources
@@ -1 +1,2 @@
-SHA512 (source-v0.39.1.tar.gz) = da8ee577aa823c9fb7b45bfd07503656561c1cd4cac42a9bf8f582f653c93b20824b3bce7bbcbd47b7f7f8e83fb1b630d6d5c983fe345c971322369f193c2c8f
+SHA512 (source-v0.42.0.tar.gz) = 839916957abc867ad9a3a32b6f74830055060b99c4c1de314f71a85f8d55ef051c5adb7c79dd7a7e10f81d1130aa032458bb65ce2454198a79faa719021f66af
+
