--- stylua-0.20.0/Cargo.toml	2024-05-29T12:17:50.878257+00:00
+++ stylua-0.20.0/Cargo.toml	2024-05-29T12:18:01.724879+00:00
@@ -23,7 +23,7 @@
 bench = false
 
 [features]
-default = ["editorconfig", "wasm-bindgen"]
+default = ["editorconfig"]
 serialize = []
 fromstr = ["strum"]
 luau = ["full_moon/roblox"]
@@ -55,9 +55,6 @@
 threadpool = "1.8.1"
 toml = "0.8.1"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-wasm-bindgen = { version = "0.2.81", optional = true }
-
 [dev-dependencies]
 criterion = "0.4.0"
 insta = { version = "1.34.0", features = ["glob"] }
@@ -75,3 +72,4 @@
 [[bench]]
 name = "docgen"
 harness = false
+
