--- du-dust-1.1.2/Cargo.toml	2025-03-01T01:06:10.172696+00:00
+++ du-dust-1.1.2/Cargo.toml	2025-03-01T01:07:47.585281+00:00
@@ -43,13 +43,6 @@
 sysinfo = "0.27"
 ctrlc = "3.4"
 chrono = "0.4"
-
-[target.'cfg(not(target_has_atomic = "64"))'.dependencies]
-portable-atomic = "1.4"
-
-[target.'cfg(windows)'.dependencies]
-winapi-util = "0.1"
-filesize = "0.2.0"
 
 [dev-dependencies]
 assert_cmd = "2"
@@ -102,3 +95,4 @@
 disk space without requiring sort or head. Dust will print a maximum of one
 'Did not have permissions message'.
 """
+
